watch_service.go

```diff
@@ -45,9 +43,13 @@ type commitWithEntry struct {
 
 func (ws *watchService) watchFile(ctx context.Context, projectName, repoName, lastKnownRevision string,
 	query *Query, timeout time.Duration) <-chan *WatchResult {
```

Comment:
As we discussed before, a user is expecting more than one watch result when he/she gets a channel as the return value. So how about making this method return just `*WatchResult` (not the channel) and renaming it to `watchFileOneTime`.

Reply:
And remove the `FileWatcher` and provide `watchFile` which returns `<-chan *WatchResult` (inside the method it keeps watching)? What do you think?

Reply:
> As we discussed before, a user is expecting more than one watch result when he/she gets a channel as the return value. So how about making this method return just `*WatchResult` (not the channel) and renaming it to `watchFileOneTime`.

Yes it's, and I would make it at next PR. This PR is for `FileWatcher` only. I think we could keep it and reuse for `watchService` :)